src/components/Card.js

```diff
@@ -16,31 +25,53 @@ export default class Card {
     return cardElement;
   }
 
-  _handleDeleteCard() {
+  getId() {
+    return this._id;
+  }
+
+
+  removeCard() {
     this._cardElement.remove();
     this._cardElement = null;
   }
 
   _handleLikeIcon() {
-    this._cardElement.querySelector(".card__like-button").classList.toggle("card__like-button_active");
+    this._likeButton.classList.toggle("card__like-button_active");
+    if (this._cardElement.querySelector(".card__like-button_active")) {
+      this._api.likeCard(this.getId());
+    } else {
+      this._api.unlikeCard(this.getId());
+    }
+  }
+
+  _setLike() {
+    if (this._isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    }
   }
 
+
   _setEventListeners() {
     this._likeButton.addEventListener("mousedown", () => this._handleLikeIcon());
```

Comment:
`mousedown` should be used only for overlay handling. In any other clicks you need to use `click` otherwise when I press the right mouse button the actions is triggered


```js

  _setEventListeners() {
    this._likeButton.addEventListener("click", () => this._handleLikeIcon());
    this._binIcon.addEventListener("click", () => {
      this._handleDelete(this);
    });

    this._cardImage.addEventListener("click", () => {
      this._handleImageClick({ link: this._link, name: this._name });
    });
  }
```